vmdb/spec/requests/api/vms_spec.rb

```diff
@@ -23,1008 +33,764 @@ def app
   end
 
   context "Vm accounts subcollection" do
-    it "query VM accounts subcollection with no related accounts" do
-      basic_authorize @cfme[:user], @cfme[:password]
-
-      vm = FactoryGirl.create(:vm_vmware)
-
-      @success = run_get "#{@cfme[:vms_url]}/#{vm.id}/accounts"
+    let(:acct1) { FactoryGirl.create(:account, :vm_or_template_id => vm.id, :name => "John") }
+    let(:acct2) { FactoryGirl.create(:account, :vm_or_template_id => vm.id, :name => "Jane") }
+    let(:vm_accounts_url)      { "#{vm_url(vm.id)}/accounts" }
+    let(:acct1_url)            { "#{vm_accounts_url}/#{acct1.id}" }
+    let(:acct2_url)            { "#{vm_accounts_url}/#{acct2.id}" }
+    let(:vm_accounts_url_list) { [acct1_url, acct2_url] }
+
+    context "query VM accounts subcollection with no related accounts" do
+      before do
+        api_basic_authorize
+
+        run_get vm_accounts_url
+      end
 
-      expect(@code).to eq(200)
-      expect(@result).to have_key("name")
-      expect(@result["name"]).to eq("accounts")
-      expect(@result["resources"]).to be_empty
+      include_examples "empty_query_result", :accounts
     end
 
-    it "query VM accounts subcollection with two related accounts" do
-      basic_authorize @cfme[:user], @cfme[:password]
-
-      vm = FactoryGirl.create(:vm_vmware)
-      acct1 = FactoryGirl.create(:account, :vm_or_template_id => vm.id, :name => "John")
-      acct2 = FactoryGirl.create(:account, :vm_or_template_id => vm.id, :name => "Jane")
+    context "query VM accounts subcollection with two related accounts" do
+      before do
+        api_basic_authorize
+        acct1
+        acct2
```

Comment:
Personally, as someone who hasn't seen these tests before, I have no idea what api_basic_authorize, acct1, and acct2 do.  Are they methods, lets, something else?  If we're creating things, I'd rather these methods return local or instance variables.  

Plus, from a glance, I have no idea what setup is required for the tests.  If this test is doing more setup than required, it's not clear because we're hiding that knowledge.  If this test fails, we have to go elsewhere to see what these things are doing.  

I'd prefer a setup method that returns things created by the method or just repeat ourselves in the  tests through a common before block.  As these tests become old and dated, we'll have to continue to maintain them and have to remember what this magic does each time.